processor/src/lib.rs

```diff
@@ -106,7 +106,10 @@ pub fn execute(
 
 /// Returns an iterator which allows callers to step through the execution and inspect VM state at
 /// each execution step.
-pub fn execute_iter(program: &Program, inputs: &ProgramInputs) -> VmStateIterator {
+pub fn execute_iter(
+    program: &Program,
+    inputs: &ProgramInputs,
+) -> VmStateIterator<MemAdviceProvider> {
```

Comment:
I think if we remove generic part from the `VmStateIterator` (as per previous comment) - we should be able to keep the signature unchanged here for now.

Reply:
Sounds good; onto that. I'll resolve this item so we keep it centralized on the previous discussion